File: L8GoogleMedium/LongestConsecutiveSequence.java

import java.util.HashSet;

public class LongestConsecutiveSequence {

    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }
        int longest = 0;
        for (int num : nums) {
            if (!set.contains(num - 1)) {
                int current = num;
                int count = 1;
                while (set.contains(current + 1)) {
                    current++;
                    count++;
                }
                longest = Math.max(longest, count);
            }
        }
        return longest;
    }

    public static void main(String[] args) {
        LongestConsecutiveSequence s = new LongestConsecutiveSequence();

        int[] nums1 = {100, 4, 200, 1, 3, 2};
        int expected1 = 4;
        int actual1 = s.longestConsecutive(nums1);
        System.out.println("Length of the longest consecutive sequence: " + actual1);
        if (actual1 == expected1) {
            System.out.println("Test case 1 passed!");
        } else {
            System.out.println("Test case 1 failed!");
        }
        System.out.println();

        int[] nums2 = {0, 3, 7, 2, 5, 8, 4, 6, 0, 1};
        int expected2 = 9;
        int actual2 = s.longestConsecutive(nums2);
        System.out.println("Length of the longest consecutive sequence: " + actual2);
        if (actual2 == expected2) {
            System.out.println("Test case 2 passed!");
        } else {
            System.out.println("Test case 2 failed!");
        }
        System.out.println();

        int[] nums3 = {1, 2, 0, 1};
        int expected3 = 3;
        int actual3 = s.longestConsecutive(nums3);
        System.out.println("Length of the longest consecutive sequence: " + actual3);
        if (actual3 == expected3) {
            System.out.println("Test case 3 passed!");
        } else {
            System.out.println("Test case 3 failed!");
        }
    }
}

File: L8GoogleMedium/LongestConsecutiveSequence.cpp
#include <iostream>
#include <unordered_set>
#include <vector>

class LongestConsecutiveSequence
{
public:
    int longestConsecutive(const std::vector<int> &nums)
    {
        std::unordered_set<int> set(nums.begin(), nums.end());
        int longest = 0;

        for (int num : nums)
        {
            if (set.find(num - 1) == set.end())
            {
                int current = num;
                int count = 1;

                while (set.find(current + 1) != set.end())
                {
                    current++;
                    count++;
                }
                longest = std::max(longest, count);
            }
        }
        return longest;
    }
};

int main()
{
    LongestConsecutiveSequence s;

    std::vector<int> nums1 = {100, 4, 200, 1, 3, 2};
    int expected1 = 4;
    int actual1 = s.longestConsecutive(nums1);
    std::cout << "Length of the longest consecutive sequence: " << actual1 << std::endl;
    if (actual1 == expected1)
    {
        std::cout << "Test case 1 passed!" << std::endl;
    }
    else
    {
        std::cout << "Test case 1 failed!" << std::endl;
    }
    std::cout << std::endl;

    std::vector<int> nums2 = {0, 3, 7, 2, 5, 8, 4, 6, 0, 1};
    int expected2 = 9;
    int actual2 = s.longestConsecutive(nums2);
    std::cout << "Length of the longest consecutive sequence: " << actual2 << std::endl;
    if (actual2 == expected2)
    {
        std::cout << "Test case 2 passed!" << std::endl;
    }
    else
    {
        std::cout << "Test case 2 failed!" << std::endl;
    }
    std::cout << std::endl;

    std::vector<int> nums3 = {1, 2, 0, 1};
    int expected3 = 3;
    int actual3 = s.longestConsecutive(nums3);
    std::cout << "Length of the longest consecutive sequence: " << actual3 << std::endl;
    if (actual3 == expected3)
    {
        std::cout << "Test case 3 passed!" << std::endl;
    }
    else
    {
        std::cout << "Test case 3 failed!" << std::endl;
    }
    std::cout << std::endl;

    return 0;
}
